securedrop_client/logic.py

```diff
@@ -341,7 +341,7 @@ def on_authenticate(self, result):
         """
         Handles the result of an authentication call against the API.
         """
-        if isinstance(result, bool) and result:
+        if not isinstance(result, Exception):  # `None` indicates successful authentication
```

Comment:
if we remove this comment then this PR can be merged regardless of the status of https://github.com/freedomofpress/securedrop-sdk/pull/74